clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackOrchestrationV1Provider.groovy

```diff
@@ -43,16 +44,23 @@ class OpenstackOrchestrationV1Provider implements OpenstackOrchestrationProvider
         .files(subtemplate)
         .disableRollback(disableRollback)
         .timeoutMins(timeoutMins)
+        .tags(tags.join(","))
         .build()
       getRegionClient(region).heat().stacks().create(create)
     }
   }
 
   @Override
-  void updateStack(String region, String stackName, String stackId, String template, Map<String, String> subtemplate, ServerGroupParameters parameters) {
+  void updateStack(String region, String stackName, String stackId, String template, Map<String, String> subtemplate,
+                   ServerGroupParameters parameters, List<String> tags) {
     handleRequest {
       Map<String, String> params = parameters.toParamsMap()
-      StackUpdate update = Builders.stackUpdate().template(template).files(subtemplate).parameters(params).build()
+      StackUpdate update = Builders.stackUpdate()
+        .template(template)
+        .files(subtemplate)
+        .parameters(params)
+        .tags(tags ? tags.join(",") : null)
```

Comment:
It would be nicer if Openstack4J took a list instead of a comma delimeted string :(